src/fs/dvfs/dvfs_util.c

```diff
@@ -148,12 +171,26 @@ int dvfs_destroy_dentry(struct dentry *dentry) {
 		if (dentry->d_inode)
 			dvfs_destroy_inode(dentry->d_inode);
 		dentry->parent->usage_count--;
+		dlist_del(&dentry->children_lnk);
```

Comment:
resource manage actions are better in one function.
I can't guarantie that this 4 lines are correct without explicit initialization/deinitialization